api/server/extension_points.go

```diff
@@ -41,9 +40,8 @@ func (s *Server) apiRouteHandlerWrapperFunc(apiHandler fnext.ApiRouteHandler) gi
 	return func(c *gin.Context) {
 		context := c.Request.Context()
 		// get the app
-		appIDorName := c.Param(api.CApp)
-		app := &models.App{Name: appIDorName, ID: appIDorName}
-		app, err := s.datastore.GetApp(context, app)
+		appName := c.Param(api.CApp)
+		app, err := s.datastore.GetAppByName(context, appName)
```

Comment:
need to use app id here for this lookup too (see comment about middleware above)